lambda_code/src/fhir_controller.py

```diff
@@ -99,23 +105,80 @@ def delete_immunization(self, aws_event):
             return self.create_response(500, unhandled_error.to_operation_outcome())
 
     def search_immunizations(self, aws_event) -> dict:
+        http_method = aws_event.get("httpMethod")
+        nhsNumberValue = None
+        diseaseTypeValue = None
+        nhsNumberParam = "-nhsNumber"
+        diseaseTypeParam = "-diseaseType"
+        if http_method == "POST":
+            # Check if the Content-Type is application/x-www-form-urlencoded
+            content_type = aws_event.get("headers", {}).get("Content-Type")
+            if content_type == "application/x-www-form-urlencoded":
+                # Parse the body of the request
+                body = aws_event["body"]
+                # Decode the base64-encoded body
+                decoded_body = base64.b64decode(body).decode("utf-8")
+                # Parse the decoded body as application/x-www-form-urlencoded
+                parsed_body = urllib.parse.parse_qs(decoded_body)
+                # Access individual form parameters and check if nhsNumber or diseaseType is present
+                nhsNumber_list = parsed_body.get(nhsNumberParam)
+                if nhsNumber_list:
+                    nhsNumberValue = nhsNumber_list[0]
+
+                diseaseType_list = parsed_body.get(diseaseTypeParam)
+                if diseaseType_list:
+                    diseaseTypeValue = diseaseType_list[0]
+                # Continue processing the request
         params = aws_event["queryStringParameters"]
-
-        if "nhsNumber" not in params:
-            return self._create_bad_request("Query parameter 'nhsNumber' is mandatory")
-        if "diseaseType" not in params:
-            return self._create_bad_request("Query parameter 'diseaseType' is mandatory")
-
-        result = self.fhir_service.search_immunizations(params["nhsNumber"], params["diseaseType"])
-
+        # If nhsNumber was not present in body then it should be in params irresepective of GET/POST
+        if nhsNumberValue is None:
+            if params is None or nhsNumberParam not in params:
+                return self._create_bad_request(
+                    f"Search Parameter {nhsNumberParam} is mandatory"
+                )
+            else:
+                nhsNumberValue = params[nhsNumberParam]
+        else:
+            #If NhsNumber was present in body as well as in QueryParams, then both values should be same
+            if nhsNumberParam in params:
+                if nhsNumberValue!=params[nhsNumberParam]:
+                    return self._create_bad_request(
```

Comment:
This is a perfectly valid request by the FHIR docs as far as I can see. In this case it will return no result. Let's leave it for now though as I think we need a separate ticket for sorting out the broken param handling and actual searching because we don't have the details of all the params.

Reply:
We have to return empty Bundle and for that we need to return some self url, what should be the value of those different params in that?
 Should it just be empty or some system generated Incorrect value?